src/Container/DispatchMiddlewareFactory.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Container;
+
+use Psr\Container\ContainerInterface;
+use Zend\Expressive\Middleware\DispatchMiddleware;
+
+class DispatchMiddlewareFactory
+{
+    public function __invoke(ContainerInterface $container) : DispatchMiddleware
+    {
+        return new DispatchMiddleware();
```

Comment:
Do we need that factory? Isn't it `invokable`?

Reply:
It _is_ invokable, but it may not always be. Additionally, having it here simplifies how we refer to it when writing configuration for different container types.